src/components/Button/src/button.module.css

```diff
@@ -0,0 +1,94 @@
+.denhaag-button {
+  font-family: var(--denhaag-font-family);
+  border-radius: var(--denhaag-border-radius);
+}
+
+.denhaag-button--focus:after, .denhaag-button:focus:after {
+  content: '';
+  display: block;
+  position: absolute;
+  top: -2px;
+  bottom: -2px;
+  left: -2px;
+  right: -2px;
+  border: var(--denhaag-focus-border);
+  border-radius: var(--denhaag-border-radius);
+}
+
+/* Due to the accessibility focus outline overlapping with the border, the outline needs to be repositioned for the border variant */
+.denhaag-button--secondary-action.denhaag-button--focus:after, .denhaag-button--secondary-action:focus:after {
+  top: -3px;
+  bottom: -3px;
+  left: -3px;
+  right: -3px;
+}
+
+.denhaag-button--primary-action {
+  background-color: var(--denhaag-button-primary-action-background-color);
+  color: var(--denhaag-button-primary-action-color);
+}
+
+.denhaag-button--primary-action.denhaag-button--hover, .denhaag-button--primary-action:hover {
+  background-color: var(--denhaag-button-primary-action-hover-background-color);
+  color: var(--denhaag-button-primary-action-hover-color);
+}
+
+.denhaag-button--primary-action.denhaag-button--disabled {
+  background-color: var(--denhaag-button-primary-action-disabled-background-color);
+  color: var(--denhaag-button-primary-action-disabled-color);
+}
+
+.denhaag-button--secondary-action {
+  background-color: var(--denhaag-button-secondary-action-background-color);
+  color: var(--denhaag-button-secondary-action-color);
+  border-width: var(--denhaag-button-border-width);
+  border-style: solid;
+  border-color: var(--denhaag-button-secondary-action-border-color);
+}
+
+.denhaag-button--secondary-action.denhaag-button--hover, .denhaag-button--secondary-action:hover {
+  background-color: var(--denhaag-button-secondary-action-hover-background-color);
+  color: var(--denhaag-button-secondary-action-hover-color);
+  border-color: var(--denhaag-button-secondary-action-hover-border-color);
+}
+
+.denhaag-button--secondary-action.denhaag-button--disabled {
+  background-color: var(--denhaag-button-secondary-action-disabled-background-color);
+  color: var(--denhaag-button-secondary-action-disabled-color);
+  border-color: var(--denhaag-button-secondary-action-disabled-border-color);
+}
+
+/* Size specific styling */
+.denhaag-button--medium {
+  padding-inline: var(--denhaag-button-medium-size-padding-inline);
+  padding-block: var(--denhaag-button-medium-size-padding-block);
+  font-size: var(--denhaag-fontsize-base);
+
+}
+
+.denhaag-button--medium.denhaag-button--secondary-action {
+  padding-inline: calc(var(--denhaag-button-medium-size-padding-inline) - var(--denhaag-button-border-width));
+  padding-block: calc(var(--denhaag-button-medium-size-padding-block) - var(--denhaag-button-border-width));
+}
+
+.denhaag-button--large {
+  padding-inline: var(--denhaag-button-large-size-padding-inline);
+  padding-block: var(--denhaag-button-large-size-padding-block);
```

Comment:
With this code, teams that don't need a different padding for all the various button sizes, still are required to specify the `--denhaag-button-large-size-padding-inline` token. Providing the regular padding as fallback value would make it much more convenient to get started and reduce maintenance. Another benefit of this approach is that introducing new variables wouldn't be a breaking change.

For example:

```css
.denhaag-button--large {
   padding-inline: var(--denhaag-button-large-size-padding-inline, var(--denhaag-button-padding-inline));
   padding-block: var(--denhaag-button-large-size-padding-block, var(--denhaag-button-padding-block));
}
```

Reply:
That's a nice touch! Thanks for the suggestion, I'll be sure to add that.